core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedClusterMessage.scala

```diff
@@ -124,6 +124,10 @@ private[spark] object CoarseGrainedClusterMessages {
 
   case class RegisterClusterManager(am: RpcEndpointRef) extends CoarseGrainedClusterMessage
 
+  // Send AM container information to the driver for Yarn client mode
+  case class YarnAmInfo(time: Long, hostName: String, logUrlInfo: Map[String, String])
```

Comment:
I'd rather not have this yarn specific either, perhaps we can come up with something more generic that could potentially be reused if other cluster manager have something similar.

Reply:
Added the generic info that can work with other Cluster Manager